monai/transforms/transforms.py

```diff
@@ -877,8 +877,75 @@ def __call__(self, img):
         assert np.all(max_end[:sd] >= self.roi_end[:sd]), 'roi end out of image space.'
 
         slices = [slice(None)] + [slice(s, e) for s, e in zip(self.roi_start[:sd], self.roi_end[:sd])]
-        data = img[tuple(slices)].copy()
-        return data
+        data = img[tuple(slices)]
+        return data.copy() if self.copy else data
+
+
+class CenterSpatialCrop:
+    """
+    Crop at the center of image with specified ROI size.
+
+    Args:
+        roi_size (list, tuple): the spatial size of the crop region e.g. [224,224,128]
+    """
+
+    def __init__(self, roi_size):
+        self.roi_size = roi_size
+
+    def __call__(self, img):
+        center = [i // 2 for i in img.shape[1:]]
+        cropper = SpatialCrop(roi_center=center, roi_size=self.roi_size, copy=False)
+        return cropper(img)
+
+
+class RandCenterSpatialCrop(Randomizable):
+    """
+    Crop at a random position in the image with the specified ROI size.
+
+    Args:
+        roi_size (list, tuple): the spatial size of the crop region e.g. [224,224,128]
+    """
+
+    def __init__(self, roi_size):
+        self.roi_size = (None,) + tuple(roi_size)
+
+        self._slices = None
+
+    def randomize(self, image_shape, roi_size):
+        self._slices = get_random_patch(image_shape, roi_size, self.R)
+
+    def __call__(self, img):
+        roi_size = get_valid_patch_size(img.shape, self.roi_size)
+        self.randomize(img.shape, roi_size)
+        return img[self._slices]
+
+
+class RandSizeSpatialCrop(Randomizable):
+    """
+    Crop image with random size ROI. It can crop at a random position as center
+    or at the image center. And allows to set the minimum size to limit the randomly
+    generated ROI. Suppose all the expected fields specified by `keys` have same shape.
+
+    Args:
+        min_roi_size (list, tuple): the spatial size of the minimum crop region e.g. [224,224,128]
+        random_center (bool): crop at random position as center or the image center.
+    """
+
+    def __init__(self, min_roi_size, random_center=True):
+        self.min_size = min_roi_size
+        self.random_center = random_center
+
+    def randomize(self, img_size):
+        min_size = [self.min_size] * len(img_size) if not isinstance(self.min_size, (list, tuple)) else self.min_size
+        self.roi_size = [self.R.randint(low=min_size[i], high=img_size[i] + 1) for i in range(len(img_size))]
+
+    def __call__(self, img):
+        self.randomize(img.shape[1:])
+        if self.random_center:
+            cropper = RandCenterSpatialCrop(self.roi_size)
```

Comment:
this transform already covers the random location cases, perhaps we should merge this class and `randcenterspatialcrop` into a `randspatialcrop`? with boolean options of `random_centre` and `random_size`

Reply:
I think users usually use RandLocSpatialCrop only and it's more clear to separate into smaller transforms for better API user experience and future extension.
Welcome a further discussion if necessary.
Thanks.